Update the README.md (#113)
vtex-apps/search-result: README.md

### Configuration

Through the Storefront, you can change the search-result behavior and interface. However, you also can make in your theme app, as Dreamstore does.

| Prop name          | Type       | Description   | Default value |
| ------------------ | ---------- | --------------------------------------------------------------------------- | --- |
| `querySchema`                      | `QuerySchema`   | Query made when there's no context                     | N/A
| `hiddenFacets`          | `HiddenFacets`  | Indicates which facets will be hidden                             | N/A
| `pagination`                   | `String`  | Pagination type (values: 'show-more' or 'infinite-scroll')       | `infinity-scroll`

QuerySchema

| Prop name          | Type       | Description  | Default value |
| ------------------ | ---------- | --------------------------------------------------------------------------- | --- |
| `maxItemsPerPage`   | `Number`   | Maximum number of items per search page                     | 10 |

> | `queryField`          | `String`  | Query field                             | N/A

Reviewer: For conciseness use:

```suggestion
| `queryField`          | `String`  | Query field                             | -
```

# VTEX Search Result

## Description

> The VTEX Search Result app is a store component that handles the result of a search, and this app is used by store theme.

Reviewer 2: ```suggestion
The VTEX Search Result app is a store component that handles with the result of our [_Search API_](https://documenter.getpostman.com/view/845/vtex-search-api/Hs43#8b71745e-00f9-6c98-b776-f4468ecb7a5e), and this app is used by store theme.
```

We add the search-result as a block in our [Store](https://github.com/vtex-apps/store/blob/master/store/interfaces.json).

To configure or customize this app, you need to import it in your dependencies in `manifest.json`.

```json
  dependencies: {
    "vtex.search-result": "3.x"
  }
```

Then, add `search-result` block into your app theme as we do in our [Store theme app](https://github.com/vtex-apps/store-theme/blob/master/store/blocks.json). 

Now, you can change the behavior of the search result block that is in the store header. See an example of how to configure: 

```json
  "search-result#department": {
    "blocks": [
      "filter-navigator",
      "gallery",
      "not-found",
      "breadcrumb",
      "order-by",
      "total-products"
    ],
    "props": {
      "querySchema": {
        "maxItemsPerPage": 2,
        "orderByField": "OrderByReleaseDateDESC"
      },
      "hiddenFacets": {
        "layoutMode1": "normal",
        "layoutMode2": "small",
        "specificationFilters": {
          "hiddenFilters": [
            {

>               "__editorItemTitle": "editor.search-result.hiddenFacets.specificationFilters.hiddenFilter"

Reviewer 2: remove this. It is just used by Storefront. If you don't add an example of `hiddenFilters`, remove it

            }
          ]
        }
      },

>       "summary": {

Reviewer 2: remove this too.

        "isOneClickBuy": false,
        "showBadge": true,
        "displayBuyButton": "displayButtonAlways",
        "showCollections": false,
        "labelSellingPrice": null,
        "showListPrice": true,
        "showLabels": true,
        "showInstallments": false,
        "showSavings": false,
        "name": {
          "showBrandName": false,
          "showSku": false,
          "showProductReference": false
        }
      },
      "pagination": "show-more"
    }
  },
```

### Blocks API

When implementing this app as a block, various inner blocks may be available. The following interface lists the available blocks within search result and describes if they are required or optional.

```json
"search-result": {
    "allowed": [
      "not-found",
      "breadcrumb",
      "filter-navigator",
      "total-products",
      "order-by"
    ],
    "required": [
      "gallery"
    ],
    "component": "index"
  },
```

> The search-result has as a required block the `gallery`. So, any search-result block implementation created must add a gallery as a block that is inside of search-result. Similarly, `gallery` has its own inner block structure that can be configured that you can see below.

Reviewer: ```suggestion
The search-result has as a required block the `gallery`. So, any search-result block implementation created must add a gallery as a block that is inside of search-result. Similarly, `gallery` has its own inner block structure that can be configured like you can see below.
```

| `pagination`                   | `String`  | Pagination type (values: 'show-more' or 'infinite-scroll')       | `infinity-scroll`

QuerySchema

| Prop name          | Type       | Description  | Default value |
| ------------------ | ---------- | --------------------------------------------------------------------------- | --- |
| `maxItemsPerPage`   | `Number`   | Maximum number of items per search page                     | 10 |
| `queryField`          | `String`  | Query field                             | N/A
| `mapField`              | `String`  | Map field                                                  | N/A |
| `restField`                   | `String`  | Other Query Strings       | N/A |
| `orderByField`                   | `String`  | Order by field (values: 'OrderByTopSaleDESC', 'OrderByReleaseDateDESC', 'OrderByBestDiscountDESC', 'OrderByPriceDESC', 'OrderByPriceASC', 'OrderByNameASC' or 'OrderByNameDESC')       | `OrderByReleaseDateDESC` |
 
HiddenFacets

| Prop name          | Type       | Description   | Default value |
| ------------------ | ---------- | --------------------------------------------------------------------------- | --- |

> | `layoutMode1`            | `String`  | Layout mode of the switcher (values: 'normal', 'small' or 'inline')    | `normal` |

Reviewer: Isn't this a `Enum` type?

Author: No, it is a String. :smile:

Reviewer: Isn't is a `Enum` of type String?

In this case 
```suggestion
| `layoutMode1`            | `Enum`  | Layout mode of the switcher (Possible values: 'normal', 'small' or 'inline')    | `normal` |
```

| `specificationFilters`   | `SpecificationFilters`  | Hide Specifications filters       | N/A |

SpecificationFilters

| Prop name          | Type       | Description  | Default value |
| ------------------ | ---------- | --------------------------------------------------------------------------- | --- |
| `hideAll`            | `Boolean`  | Hide specifications filters    | false |
| `hiddenFilters`            | `Array(String)`  | Array of specifications filters that should be hidden  | N/A |

Also, you can configure the product summary that is defined on search-result. See [here](https://github.com/vtex-apps/product-summary/blob/master/README.md#configuration) the Product Summary API. 

### Styles API

This app provides some CSS classes as an API for style customization.

To use this CSS API, you must add the `styles` builder and create an app styling CSS file.

1. Add the `styles` builder to your `manifest.json`:

```json
  "builders": {
    "styles": "1.x"
  }
```

2. Create a file called `vtex.searchResult.css` inside the `styles/css` folder. Add your custom styles:

```css
.container {
  margin-top: 10px;
}
```

#### CSS namespaces

> Below, we describe the namespaces that are defined in the minicart.

Reviewer 2: search-result

# VTEX Search Result

## Description

> The VTEX Search Result app handles the result of a search and this app is used by the Dreamstore product.

Reviewer 2: The VTEX Search Result app is a store component that handles ...

Reviewer 2: ... by store theme.

Please check the last doc of the Minicart

| `specificationFilters`   | `SpecificationFilters`  | Hide Specifications filters       |

SpecificationFilters

| Prop name          | Type       | Description                                                                 |
| ------------------ | ---------- | --------------------------------------------------------------------------- |
| `hideAll`            | `Boolean`  | Hide specifications filters    |
| `hiddenFilters`            | `Array(String)`  | Array of specifications filters that should be hidden  |

Also, you can configure the product summary that is defined on minicart. See [here](https://github.com/vtex-apps/product-summary/blob/master/README.md#configuration) the Product Summary API. 

### Styles API
This app provides some CSS classes as an API for style customization.

| Token name         | Component          | Description                                            |
| ------------------ | ----------         |------------------------------------------------------- |
| `container`        | [SearchResult](https://github.com/vtex-apps/search-result/blob/master/react/components/SearchResult.js) | The main container of search-result |

> | `buttonShowMore`        | [ShowMoreLoaderResult](https://github.com/vtex-apps/search-result/blob/master/react/components/loaders/ShowMoreLoaderResult.js) | Show the see more button |

Reviewer: The description, in this case, can't be an action 

```suggestion
| `buttonShowMore`        | [ShowMoreLoaderResult](https://github.com/vtex-apps/search-result/blob/master/react/components/loaders/ShowMoreLoaderResult.js) | The "Show more" button in ... |
```

| `hideAll`            | `Boolean`  | Hide specifications filters    | false |
| `hiddenFilters`            | `Array(String)`  | Array of specifications filters that should be hidden  | `undefined` |

Also, you can configure the product summary that is defined on search-result. See [here](https://github.com/vtex-apps/product-summary/blob/master/README.md#configuration) the Product Summary API. 

### Styles API

This app provides some CSS classes as an API for style customization.

To use this CSS API, you must add the `styles` builder and create an app styling CSS file.

1. Add the `styles` builder to your `manifest.json`:

```json
  "builders": {
    "styles": "1.x"
  }
```

2. Create a file called `vtex.searchResult.css` inside the `styles/css` folder. Add your custom styles:

```css
.container {
  margin-top: 10px;
}
```

#### CSS namespaces

Below, we describe the namespaces that are defined in the minicart.

| Token name         | Description                   | Component Source |
| ------------------ | ----------         |------------------------------------------------------- |

> | `container`         | The main container of search-result | [SearchResult](https://github.com/vtex-apps/search-result/blob/master/react/components/SearchResult.js) |

Reviewer: You can use only "/react/componente/SearchResult.js" on Thais links
